Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -114,8 +115,33 @@ export class ArtifactEngine {
         }
     }
 
+    shouldDownload(path: string) {
+        var shouldDownload = false;
+        for (let pattern of this.patternList) {
+            if (pattern.charAt(0) === '-' && (minimatch(path, pattern.substr(1), { dot: true, nocase: true }))) {
+                return false;
+            }
+            else {
+                if (minimatch(path, (pattern.charAt(0) === '+') ? pattern.substr(1) : pattern, { dot: true, nocase: true })) {
+                    shouldDownload = true;
+                }
+            }
+        }
+        return shouldDownload;
+    }
+
+    createPatternList(artifactEngineOptions: ArtifactEngineOptions) {
+        if (!artifactEngineOptions.itemPattern) {
+            this.patternList = ['**'];
+        }
+        else {
+            this.patternList = artifactEngineOptions.itemPattern.split('","').map((pattern) => pattern.replace(/"/g, ''));
```

Comment:
can pattern have space after comma? like:
"+drop\starstar", "-drop\start.exe" (notice space after comma)

Reply:
We planned the pattern string to be only as "+drop\**","-drop\folder\**" (without spaces). Should I add support for spaces also?